doc/metadata-server/README.md

```diff
@@ -0,0 +1,327 @@
+    && checkSignature txMetadata.payload pubKey MarloweTxMetadata.signature
+      then
+        Just MarloweTxMetadata.metadata
+      else
+        Nothing
+```
+
+> In the current implementation `MarloweParams` are used to derive the final validator. `MarloweParams` provide "low level" details (the currency symbol) which are used to implement roles handling on chain. From the perspective of the contract semantics they are not important. We treat them as implementation detail here hidden behind `checkValidator` but in the final implementation spec we should also handle them somehow.
+> It is worth to mention that in the context of [CIP-33](https://github.com/cardano-foundation/CIPs/tree/master/CIP-0033) it seems beneficial to move `MarloweParams` to the `datum` so the validator script could be shared and the transaction size could be minimized.
+
+It may seem a bit artificial or unrealistic that we want to delegate submission process to some third parties and not rely on the submitter of the `MarloweTx` as the author of metadata attachments. On the other hand there are some cases where this separation seems to be useful:
+
+- Marlowe Marketplace can possibly perform submissions of paid and certified contracts on behalve of the user. We may want to perform creation of these transactions by the platform so we can confirm to others that we really give a credit to a given transaction and we associate it with certified contract template. User should pass to us contract parameters together with signed `MarloweTxMetadata` (which contains the required triple) so we can still proove to the others that a given `MarloweTx` was created and described by a particular party.
+
+  Of course we are not able to prevent users from using certified / paid templates on their own (without paying fee) because in some sens all Marlowe contracts are open sourced (at least through inspection of the `Tx`). In such a case we are not going to endorse these transactions on-chain and on the platform.
+
+- In multiparty agreement it can be not so important who is the owner of the initial transaction (contract participant, some third party auditor, layer etc.). On the other hand it is rather important that the attachment for a transaction was really dedicated for its semantic meaning.
+
+- Using these schema we can require distribution of role tokens to a specific, generic utility scripts with constant addresses (like "deposit scripts") and provide the same audited Marlowe contract with the same role distribution which can be safely reused and certified. By using this schema we can push a possible role distribution logic outside of our Marlowe interpreter / validator (we don't have to change our implementation) to role scripts (with single and simpler to audit responsiblity) and distribute reusable instances of Marlowe contracts together with verified set of metadata (authorized for this Contract and hard coded role to scripts distribution).
+
+## Extended Marlowe
```

Comment:
In the overview we could say why we are considering contracts in Marlowe and Extended Marlowe separately. Also can we generalise this so that it would be possible to use other kinds of templates (like a Haskell function, for example)?